src/kudu/consensus/consensus_peers.cc

```diff
@@ -206,8 +210,28 @@ Status Peer::SignalRequest(bool even_if_queue_empty) {
   return Status::OK();
 }
 
+bool Peer::ProxyBatchDurationHasPassed() {
+  const std::string& uuid = peer_pb_.permanent_uuid();
+  std::string next_hop_uuid;
+
+  // TODO: The routing table is consulted again in RequestForPeer(), ideally we
+  // can do it once
+  queue_->GetNextRoutingHopFromLeader(uuid, &next_hop_uuid);
```

Comment:
is this causing some new lock in the SendNextRequest code path?

Reply:
Yes, we're locking the routing table in read mode to get the hop info.